semgrep/semgrep/metric_manager.py

```diff
@@ -44,7 +48,28 @@ def __init__(self) -> None:
         self._file_stats: List[Dict[str, Any]] = []
         self._rule_stats: List[Dict[str, Any]] = []
 
-        self._send_metrics = False
+        self._send_metrics: MetricsState = MetricsState.OFF
+        self.using_server = False
```

Comment:
For consistency with the rest of this class, can we make this internal only and expose a setter.